### PR TITLE
Changes to Timeseries Aggregates

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

- Makes separate queries for aggregates  
- Makes separate queries for totals  
- Alters how dates filters are made on the backend  
- Alters how per-user calculations are done a bit

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

**Comment by michielderoos** [comment]
[Alter Aggregation Strategy for metrics to be just the time-bound stretched version of individual TS datapoints](https://trello.com/c/UVHbkX9N/479-alter-aggregation-strategy-for-metrics-to-be-just-the-time-bound-stretched-version-of-individual-ts-datapoints)

File: app/client/components/filterModule/FilterModule.jsx [hunk 1]
```diff
@@ -94,11 +94,13 @@ class FilterModule extends React.Component {
     if (groupBy) {
       params.group_by = groupBy;
     }
-
-    if (dateRange) {
+    if (dateRange && dateRange[0]) {
```
Comment (michielderoos) [inline app/client/components/filterModule/FilterModule.jsx:97]: Just pasted from this PR https://github.com/teamsempo/SempoBlockchain/pull/410

Comment (tristanhcole) [reply app/client/components/filterModule/FilterModule.jsx:97]: Does that close #410 ?

File: app/server/utils/metrics/metric.py [hunk 2]
```diff
@@ -6,63 +6,122 @@
...
 class Metric(object):
-    def execute_query(self, user_filters: dict = None, date_filters=None, enable_caching=True, population_query_result=False):
-        user_filters = user_filters or {}
-        # Apply stock filters
-        filtered_query = self.query
-        for f in self.stock_filters:
-            filtered_query = filtered_query.filter(*f)
+    def execute_query(self, user_filters: dict = None, date_filters_dict=None, enable_caching=True, population_query_result=False, calculate_only_aggregates=False):
+        """
+        :param user_filters: list of filters to apply to all metrics
```
Comment (enjeyw) [inline app/server/utils/metrics/metric.py:11]: nit: typehint is a dict but this says list?

File: app/server/utils/metrics/metric.py [hunk 3]
```diff
@@ -6,63 +6,122 @@
...
+                    filtered_query = filtered_query.filter(*date_filters)
 
-        if not self.bypass_user_filters:
-            filtered_query = filters.apply_filters(filtered_query, user_filters, self.object_model)
+            if not self.bypass_user_filters:
+                filtered_query = filters.apply_filters(filtered_query, user_filters, self.object_model)
 
-        result = metrics_cache.execute_with_partial_history_cache(self.metric_name, filtered_query, self.object_model, self.caching_combinatory_strategy, enable_caching) 
-        if not self.timeseries_actions:
+            result = metrics_cache.execute_with_partial_history_cache(self.metric_name, filtered_query, self.object_model, self.caching_combinatory_strategy, enable_caching)
```
Comment (enjeyw) [inline app/server/utils/metrics/metric.py:52]: nit: Can we try keep line lengths to <120 chars? (few places like this) black.py is a todo!

File: app/server/utils/metrics/metric.py [hunk 4]
```diff
@@ -6,63 +6,122 @@
...
+            if not self.bypass_user_filters:
+                filtered_query = filters.apply_filters(filtered_query, user_filters, self.object_model)
 
-        result = metrics_cache.execute_with_partial_history_cache(self.metric_name, filtered_query, self.object_model, self.caching_combinatory_strategy, enable_caching) 
-        if not self.timeseries_actions:
+            result = metrics_cache.execute_with_partial_history_cache(self.metric_name, filtered_query, self.object_model, self.caching_combinatory_strategy, enable_caching) 
+            if not actions[query]:
+                results[query] = result
+            else:
+                results[query] = process_timeseries.process_timeseries(result, population_query_result, actions[query])
```
Comment (enjeyw) [inline app/server/utils/metrics/metric.py:56]: Nit: This method should is no longer just applicable to timeseries (also applies to aggregates)

**Pull request merged**